src/lib.rs

```diff
@@ -151,6 +174,47 @@ impl_fft_with_opts_and_plan_for!(
     16
 );
 
+/// Utility function to separate interleaved format signals (i.e., Vector of Complex Number Structs)
+/// into separate vectors for the corresponding real and imaginary components.
+pub fn separate_re_im<T: Float>(signal: &[Complex<T>], chunk_size: usize) -> (Vec<T>, Vec<T>) {
+    let mut reals = Vec::with_capacity(signal.len());
+    let mut imaginaries = Vec::with_capacity(signal.len());
+    let iter = signal.chunks_exact(chunk_size);
+    let rem = iter.remainder();
+
+    // We don't assume power of 2 size, so we don't use `chunks_exact`.
+    // Process each chunk, including the last chunk which may be smaller.
+    for chunk in iter {
+        for num in chunk {
+            reals.push(num.re);
+            imaginaries.push(num.im);
+        }
+    }
```

Comment:
`chunks_exact_mut` is actually perfectly fine as long as you also handle the `.remainder()` separately. It's way faster because it's easy for the compiler to vectorize.

But I have a better idea: IIRC `zune-image` has deinterleaving as a simple, safe iterator that vectorizes into top notch code. Just reuse that, stick `multiversion` attribute on it, and it should get way way faster.